doc/OnlineDocs/contributed_packages/mcpp.rst

```diff
@@ -0,0 +1,43 @@
+Pyomo Interface to MC++
+=======================
+
+The Pyomo-MC++ interface allows for bounding of factorable functions using the MC++ library developed by
+the OMEGA research group at Imperial College London.
+Documentation for MC++ may be found on the `MC++ website`_.
+
+.. _MC++ website: https://dx.doi.org/10.1016/0098-1354(95)00219-7
+
+Installation
+------------
+
+Support for MC++ has only been validated by Pyomo developers using Linux and OSX.
+Installation instructions for the MC++ library may be found on the `MC++ website`_.
+Until a more simple installation scheme is developed, you may install MC++ and this interface using this guide.
+
+We assume that you have installed MC++ into a directory of your choice.
+We will denote this directory by ``$MCPP_PATH``.
+On my machine, the file ``/home/qichen/.solvers/MC++/mcpp/INSTALL`` exists.
```

Comment:
I don't think you need to go in as much detail here with listing your personal directory structure.  I think this should be fairly obvious to users after following the MC++ installation steps.